community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreUpgrader.java

```diff
@@ -116,8 +118,6 @@ public void migrateIfNeeded( File storeDirectory)
             throw new UpgradeNotAllowedByConfigurationException();
         }
 
-        CustomIOConfigValidator.assertCustomIOConfigNotUsed( config, CUSTOM_IO_EXCEPTION_MESSAGE );
-
```

Comment:
It's actually `StoreUpgrader` that reads store version and passes it to `StoreMigrator`.